Git-commit: e699cce29accb713200ba94c88c5b98e2ccf4b75
From: Jens Axboe <axboe@kernel.dk>
Date: Sun, 16 Oct 2022 17:24:10 -0600
Subject: [PATCH] io_uring/rw: remove leftover debug statement
References: bsn#19
Patch-mainline: v5.10.165

commit 5c61795ea97c170347c5c4af0c159bd877b8af71 upstream.

This debug statement was never meant to go into the upstream release,
kill it off before it ends up in a release. It was just part of the
testing for the initial version of the patch.

Fixes: 2ec33a6c3cca ("io_uring/rw: ensure kiocb_end_write() is always called")
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 io_uring/io_uring.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/io_uring/io_uring.c b/io_uring/io_uring.c
index f989d699ed99..642e1a0560c6 100644
--- a/io_uring/io_uring.c
+++ b/io_uring/io_uring.c
@@ -2708,8 +2708,6 @@ static void io_req_io_end(struct io_kiocb *req)
 {
 	struct io_rw *rw = &req->rw;
 
-	WARN_ON(!in_task());
-
 	if (rw->kiocb.ki_flags & IOCB_WRITE) {
 		kiocb_end_write(req);
 		fsnotify_modify(req->file);
-- 
2.35.3

